src/body.rs

```diff
@@ -448,3 +462,33 @@ impl BufRead for Body {
         Pin::new(&mut self.reader).consume(amt)
     }
 }
+
+/// Look at first few bytes of a file to determine the mime type.
+/// This is used for various binary formats such as images and videos.
+#[cfg(feature = "async_std")]
+async fn peek_mime(file: &mut async_std::fs::File) -> io::Result<Option<Mime>> {
+    // Reading the first 8 bytes should be enough to sniff the mime type.
+    let mut buf = [0_u8; 8];
+    file.read(&mut buf).await?;
+    let mime = Mime::sniff(&buf).ok();
+
+    // Reset the file cursor back to the start.
+    file.seek(io::SeekFrom::Start(0)).await?;
+    Ok(mime)
+}
+
+/// Look at the extension of a file to determine the mime type.
+/// This is useful for plain-text formats such as HTML and CSS.
+#[cfg(feature = "async_std")]
+fn guess_ext(path: &Path) -> Option<Mime> {
+    let ext = path.extension().map(|p| p.to_str()).flatten();
+    match ext {
+        Some("html") => Some(mime::HTML),
+        Some("js") | Some("mjs") => Some(mime::JAVASCRIPT),
+        Some("json") => Some(mime::JSON),
+        Some("css") => Some(mime::CSS),
+        Some("svg") => Some(mime::SVG),
+        Some("wasm") => Some(mime::WASM),
```

Comment:
This shouldn't be necessary, as infer supports WASM detection: https://github.com/bojand/infer#application

Reply:
ah, great catch!